src/Engine/Renderer/Material/PlainMaterial.inl

```diff
@@ -0,0 +1,33 @@
+
+namespace Ra {
+namespace Engine {
+
+// Add a texture as material parameter from an already existing Radium Texture
+inline void PlainMaterial::addTexture( const TextureSemantic& semantic, Texture* texture ) {
+    m_textures[semantic] = texture;
+    // remove pendingTexture with same semantic, since the latter would
+    // overwrite the former when updateGL will be called.
+    m_pendingTextures.erase( semantic );
+}
+
+// Add a texture as material parameter with texture parameter set by the caller
+// The textures will be finalized (i.e loaded from a file if needed and transformed to OpenGL
+// texture) only when needed by the updateGL method.
```

Comment:
Why are these comments not in the header file ? They describe side effect that might be interesting for the user.